design/authorization.md

```diff
@@ -0,0 +1,85 @@
+* Dave has access to the branch `prod` content, and wants to update the content of the table `Foo`. He creates a new reference named `dave-experiment`, and executes several queries against this branch to modify table `Foo`. Each modification is a commit done against `dave-experiment` branch which is approved by Nessie server. When all the desired modifications are done, Dave attempts to merge the changes back to the `prod` branch. However Dave doesn't have the rights to modify the `prod` branch, causing Nessie to deny the request.
+
+## Access control model
+Any object in Nessie can be designated by a pair of coordinates (reference, path). It seems logical that access control is also designed around those two concepts
+
+### Reference access control
+References can be designated by name or by hash. However the practicality of using hashes for access control seems very low as those hashes are supposedly random and not known in advance. So focus will be on named references (branches and tags).
+
+Several operations can be exercised against a reference:
+* create a new named reference
+* assign a hash to a reference
+* delete a reference
+* list objects present in the tree
+* read objects content in the tree
+* commit a change against the reference
+
+### Path access control
+For a specific reference, an entity is designated by its path which is why a simple way of performing access control can be done by applying restrictions on path.
+
+Several operations can be exercised against a entity:
+* create a new entity
+* delete an entity
+* update entity's content
+
+Note that those operations combine themselves with the reference operations. For example to actually be able to update the content of an entity, user needs both permission to do the update AND to commit the change against the reference where the change will be stored
+
+### Other operations
+
+Some operations like performing garbage collection or changing access control do apply to the whole Nessie repository, and as such are control by their own specific set of permissions.
+
+
+## Server interface
+Although multiple server implementations may exist and conversely multiple access control interface may exist, this section aims to describe an interface to be used by the `JAX-RS` reference implementation of Nessie present under `servers/services`. The interface is composed of various methods for each operation to validate, which accept some access control context providing user identity and some arguments regarding the object to be accessed.
+
+More concretely, interface would look like this:
+
+    interface AccessChecker {
+      void canCreateReference(AccessContext context, NamedRef ref) throws AccessControlException;
```

Comment:
I might call these 'check' instead of 'can' since they don't return a boolean. Can sounds like a question thus expected a return value.